javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,222 @@
+------------------------------------------------------------------------------------------------ */
+
+const addExclamation = (arr) => {
+  // Solution code here...
+  let newArr = [];
+
+  arr.forEach(str => {
+    let newArray = str + '!';
+    newArr.push(newArray);
+  });
+  return newArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named `allUpperCase` that takes an array of strings, and returns a new array of the strings converted to upper case.
+
+Use `forEach` to loop over the input array. The modified strings should each be added into a local array. Return that local array.
+------------------------------------------------------------------------------------------------ */
+
+const allUpperCase = (arr) => {
+  // Solution code here...
+  let newArr = [];
+
+  arr.forEach(str => {
+    let newArray = str.toUpperCase();
+    newArr.push(newArray);
+  });
+  return newArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named `greeting` that takes in a single string and returns the string in all uppercase letters, and followed by an "!".
+
+Then, write a function named `speaker` that takes in an array of strings and a callback function. 
+
+Use `forEach` to build a new array of strings, each string modified by the callback. Return the new array. 
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => {
+  // Solution code here...
+
+  return word.toUpperCase() + '!';
+};
+
+
+const speaker = (words, callback) => {
+  // Solution code here...
+  let newArr = [];
+
+  words.forEach(arr => {
+    callback(arr);
+    newArr.push(callback(arr));
+  });
+  return newArr;
+};
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named addValues that takes in an array and a value and pushes the value into the array. This function does not need a return statement.
+
+Then, write a function named addNumbers that takes in four arguments:
+  - A number to be added to an array
+  - An array into which the number should be added
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr, value) => {
+  // Solution code here...
+  let newArr = [];
+  newArr.push(value);
+};
+
+const addNumbers = (num, arr, times, callback) => {
+  // Solution code here...
+  arr.forEach.push(num);
+
+};
```

Comment:
For `addNumbers` we have four different inputs, one of which is `times`. 

The instructions want us to 'invoke the callback as many times as necessary`.  What way can we tell this program to run `times` amount? 

Once we tell it to the number of times the number should be added `times` amount, we can invoke the callback (`callback()`). 👍